### ISSUE TITLE
[Feature] Do not report statistics for append only table or predications without keys

### ISSUE THREAD

**Issue by FangYongs** [apache/incubator-paimon#2185]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Motivation

Some olap queries in tpc-h will run faster when we perform them without statistics for append-only table or predications without keys. We can add an option for this to improve the olap latency.

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by schnappi17** [comment]
@FangYongs Please assign it to me, I'll take it, thanks~

**Comment by JingsongLi** [comment]
Hi @FangYongs , do you have some performance benchmark?

**Comment by FangYongs** [comment]
@JingsongLi Yes. We are currently using `TPC-H` for performance benchmark of `Flink` and `Paimon`, and we found statistics on the append-only table can cause some query performance to regress. When we turn off statistics separately for these queries, the query latencies are lower. So we would like to add an option in `Paimon` to uniformly support this behavior.

**Comment by JingsongLi** [comment]
@FangYongs I see, Is there some benchmark result? How large is the performance regression.
